### PR TITLE
ecc signature bugfix

### PR DESCRIPTION
**Pull request by yanxi123-com**
When I sign the data `02dd11a0cd4ec12119e2e72d1a553e657323435e08063ac511d8b7f52802cf9c`, I found error. (you can run `dart example/eosdart_ecc_example.dart`).

I fix it refer the https://github.com/EOSIO/eosjs-ecc/blob/master/src/ecdsa.js#L181

### CODE REVIEW

**Comment by fzxu** [comment]
thanks for the PR! could you keep the original test case and add a new one for this issue?

**Comment by yanxi123-com** [comment]
ok, I have revert the origin example, and add a new one.

File: lib/src/signature.dart [hunk 1]
```diff
@@ -145,7 +146,38 @@ class EOSSignature extends EOSKey {
...
+    ECPoint Q = multiplyTwo(R, s, G, eNeg) * rInv;
     return Q;
   }
+
+  static bool testBit(BigInt j, int n) {
+    return (j >> n).toUnsigned(1).toInt() == 1;
+  }
+
+  static ECPoint multiplyTwo(ECPoint t, BigInt j, ECPoint x, BigInt k) {
+    var i = max(j.bitLength, k.bitLength) - 1;
```
Comment (fzxu) [inline lib/src/signature.dart]: Could you change the var to the actual type?

**Review by fzxu — CHANGES_REQUESTED** [review]
Thanks! Could you change all the `var` to the actual type? since this project tries to explicit the type as much as possible.

The test you added is the sample, could you add a test case for it? You can put it in here: https://github.com/primes-network/eosdart_ecc/blob/master/test/signature_test.dart

File: lib/src/signature.dart (hunk 1 above)
Comment (yanxi123-com) [reply lib/src/signature.dart]: fixed.

Comment (fzxu) [reply lib/src/signature.dart]: thx. Could you move this sample to the test?

Comment (yanxi-me) [reply lib/src/signature.dart]: added `test/sign_string_test.dart` with 4 cases.

This PR fix #2. 

#3 and #4 is still failed.

**Comment by yanxi123-com** [comment]
added `test/sign_string_test.dart` with 5 cases.

```dart
    test('ECC Sign #1', () {
      var result = eccSignAndVerify('data');
      expect(result, true);
    });

    test('ECC Sign #2', () {
      var result = eccSignAndVerify(
          '02dd11a0cd4ec12119e2e72d1a553e657323435e08063ac511d8b7f52802cf9c');
      // Error: Value x must be smaller than q
      expect(result, true);
    });

    test('ECC Sign #3', () {
      var result = eccSignAndVerify('475');
      // Error: Value x must be smaller than q
      expect(result, true);
    });

    test('ECC Sign #4', () {
      var result = eccSignAndVerify('5488');
      // Error: Invalid point compression
      expect(result, true);
    });

    test('ECC Sign #5', () {
      var result = eccSignAndVerify('2097');
      // Error: nR is not a valid curve point
      expect(result, true);
    });
```

This PR fix #2. 

But #3, #4, #5 are still failed. we can continue to work on them.

**Comment by fzxu** [comment]
will you fix the test failure?

**Comment by yanxi123-com** [comment]
I found something wrong was made by dart core sdk, I have commit issue to the dart team. https://github.com/dart-lang/sdk/issues/37008

**Comment by yanxi123-com** [comment]
I have fixed the bugs in test case ugly, and I can't make sure it have no future errors, so I just made the change in my own project. We can wait for the fix of dart BigInt bug.

**Comment by fzxu** [comment]
just checking, is there any update on this one?

**Comment by yanxi123-com** [comment]
> just checking, is there any update on this one?

Dart team fixed the issue, waiting for the new SDK with the fix.
(I just tested on Dart 2.3.2, the issue is still there)

**Comment by yanxi123-com** [comment]
tested on dart 2.4.0, test/sign_string_test.dart passed.

```
➜  eosdart_ecc git:(master) dart test/sign_string_test.dart 
00:00 +0: EOS signature string tests ECC Sign #1
00:00 +1: EOS signature string tests ECC Sign #2
00:00 +2: EOS signature string tests ECC Sign #3
00:00 +3: EOS signature string tests ECC Sign #4
00:00 +4: EOS signature string tests ECC Sign #5
00:00 +5: All tests passed!
```

**Comment by fzxu** [comment]
cool! Thanks for the PR!

**Pull request merged**